app/scripts/lib/stream-utils.js

```diff
@@ -24,3 +24,13 @@ export function setupMultiplex(connectionStream) {
   });
   return mux;
 }
+
+/**
+ * Checks if a stream is usable
+ *
+ * @param {stream.Stream} stream - the stream to check
+ * @returns {boolean} if the stream can be written to
+ */
+export function isStreamWritable(stream) {
+  return stream.writable && !stream.destroyed && !stream._writableState?.ended;
+}
```

Comment:
Hm, will this return false negatives for <3.x streams? If so, we should either note that it can't be used with all streams, or check for the existence of the `writable` and/or `destroyed` properties first.

Reply:
is there something here we can use for compat https://www.npmjs.com/package/end-of-stream

Reply:
@rekmarks I have now added test-cases for expected return values for each streams library (as well as `ObjectMultiplex`) for following:
- Newly constructed stream instance
- Newly constructed stream instance; call `destroy()`
- Newly constructed stream instance; call `end()`


Any other scenario you can think of that should be captured there?

Reply:
> is there something here we can use for compat https://www.npmjs.com/package/end-of-stream

@kumavis Looks like we don't need it, unless you have some invariant in mind I haven't covered.